### PR TITLE
hardening/1048_remove_support_to_old_parameter_matching_table

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #1048 
* 100% Unit tests passed:
```
Results :
Tests run: 91, Failures: 0, Errors: 0, Skipped: 0
``` 
* Assignee: @frbattid

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -14,3 +14,4 @@
 - [cygnus] [doc] Update the sink docs regarding the naming conventions (#630)
 - [cygnus-ngsi] [hardening] Change default credentials for docker MySQL agent (#1038)
 - [cygnus-ngsi] [feature] Add MongoDB and STH sinks to docker agent (#1044)
+- [cygnus-ngsi] [hardening] Remove support to old 'matching_table' parameter (#1048)
```
Comment (frbattid) [inline CHANGES_NEXT_RELEASE]: old --> deprecated

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java [hunk 2]
```diff
@@ -174,15 +174,10 @@ public void close() {
         @Override
         public void configure(Context context) {
             String groupingRulesFileNameTmp = context.getString("grouping_rules_conf_file");
-            String matchingTableFileName = context.getString("matching_table");
             
             if (groupingRulesFileNameTmp != null && groupingRulesFileNameTmp.length() > 0) {
                 groupingRulesFileName = groupingRulesFileNameTmp;
                 LOGGER.debug("[de] Reading configuration (grouping_rules_file=" + groupingRulesFileName + ")");
```
Comment (frbattid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: [de] --> [gi]

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java [hunk 3]
```diff
@@ -174,15 +174,10 @@ public void close() {
...
-            } else if (matchingTableFileName != null && matchingTableFileName.length() > 0) {
-                groupingRulesFileName = matchingTableFileName;
-                LOGGER.debug("[de] Reading configuration (matching_table=" + groupingRulesFileName + ")"
-                        + " -- DEPRECATED, use grouping_rules_file instead");
             } else {
                 groupingRulesFileName = null;
                 LOGGER.debug("[de] Defaulting to grouping_rules_file=null");
```
Comment (frbattid) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: [de] --> [gi]

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (pcoello25) [inline CHANGES_NEXT_RELEASE]: Fixed in c418690

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java (hunk 2 above)
Comment (pcoello25) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: Fixed in c418690

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java (hunk 3 above)
Comment (pcoello25) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/interceptors/NGSIGroupingInterceptor.java]: Fixed in c418690

**Comment by frbattid** [comment]
LGTM

**Pull request merged**